File: 4-Patron/php/AdaptadorO_SkinOnline.php
<?php
class AdaptadorO_SkinOnline extends Skin
{
  private $skin;

  function __construct($skin)
  {
    $this->skin = $skin;
  }

  function cargarNuevoSkin($ruta)
  {
    $nSkin = $this->skin->getFromServer($ruta);
    $info = $nSkin[0];
    $txs = $nSkin[1];
    $this->skin = new SkinOnline($info, $txs)
    $tipoSkin = "Online";
  }

  function mostrarSkin()
  {
    $this->skin->mostrarTexturaSkin();
  }
}
?>

File: 4-Patron/python/AdaptadorO_SkinOnline.py
import ISkinLocal
import SkinOnline


class AdaptadorC_SkinLocal(ISkinLocal, Skin):
    def __init__(self):
        super().__init__()
        self.__skin = None

    def cargarSkinLocal(self, url):
        info, textura = self.getFromServer(url)
        self.__skin = SkinOnline(info, textura)
        self.__tipo = 'Online'

    def mostrarSkin(self):
        return self.__skin.mostrarTexturaSkin()
